### PR TITLE
sql/schemachanger: enable CREATE SCHEMA in declarative schema changer

### LINKED ISSUE #117648: schemachanger: Enable CREATE SEQUENCE / SCHEMA
In 23.2 development cycle, we implemented 3 things in the declarative schema changer:

- [x]  CREATE SEQUENCE (https://github.com/cockroachdb/cockroach/pull/117793)
- [ ]  CREATE SCHEMA
- [x]  ALTER TABLE with multiple commnds (https://github.com/cockroachdb/cockroach/pull/119269)

This issue tracks the effort to unfallback those three things in declarative schema changer.

Epic CRDB-31331
Jira issue: CRDB-35289

### PR DESCRIPTION
**Pull request by fqazi**
This patch enables CREATE SCHEMA in the declarative schema changer and addresses the following issues

1. Enforce restrict_system_interface for the declarative schema changer
2. Fix a dependency issue where other objects could not be created with schemas since the name wasn't populated early enough
3. Update test cases and enable support for CREATE SCHEMA in the declarative schema changer by default.

Fixes: #117648

### CODE REVIEW

File: pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go [hunk 1]
```diff
@@ -46,3 +46,20 @@ func init() {
 		},
 	)
 }
+
+func init() {
+	registerDepRule(
```
Comment (annrpom) [inline pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go:51]: nice catch!

**Review by annrpom — APPROVED** [review]
nice work! i was always curious to see a concrete example of a new deprule being added. 

we can also enable `CREATE SCHEMA AUTHORIZATION` in the schemachange workload now:

https://github.com/cockroachdb/cockroach/blob/02843bd1680167271baf6fcb2c060ca0d7d25c48/pkg/workload/schemachange/operation_generator.go#L3836-L3841

**Review by rafiss — APPROVED** [review]
lgtm! thanks for taking care of this

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/120091#-:-NsUkFcuCiu45oyUW6eB:b-tpxlxe)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @annrpom and @fqazi)

___
*[`pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go` line 60 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/120091#-NsUjqyJEBV-WtUXHqjf:-NsUjqyJEBV-WtUXHqjg:bh8njqn) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6556ddeab7a046531e5b7f36de55fbbb3e25801/pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go#L60)):*
> ```Go
> 				to.Type((*scpb.SchemaParent)(nil)),
> 				JoinOnDescID(from, to, "schema"+
> 					"-id"),
> ```

super nit: extra newline

**Review by fqazi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/120091)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @annrpom, @herkolategan, @rafiss, and @srosenberg)

___
*[`pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go` line 51 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/120091#gh-1518052743:-NsUxdEX3eGo4OLObPKW:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6556ddeab7a046531e5b7f36de55fbbb3e25801/pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go#L51)):*
<details><summary><i>Previously, annrpom (annie pompa) wrote…</i></summary><blockquote>

nice catch! 
</blockquote></details>

Done.
___
*[`pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go` line 60 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/120091#-NsUjqyJEBV-WtUXHqjf:-NsUxdSJCMoYcmyd1riy:bjb8s0n) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6556ddeab7a046531e5b7f36de55fbbb3e25801/pkg/sql/schemachanger/scplan/internal/rules/current/dep_create.go#L60)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

super nit: extra newline
</blockquote></details>

Ugh, good find

**Comment by fqazi** [comment]
@rafiss @annrpom TFTR!

bors r+

**Comment by fqazi** [comment]
bors retry

**Pull request merged**